Application/EdFi.Ods.Sandbox/Repositories/ClientAppRepo.cs

```diff
@@ -436,14 +437,31 @@ public void SetDefaultVendorOnUserFromEmailAndName(string userEmail, string user
             }
         }
 
-        public Vendor CreateOrGetVendor(string userEmail, string userName, IEnumerable<string > namespacePrefixes)
+        public async Task SetDefaultVendorOnUserFromEmailAndNameAsync(string userEmail, string userName, IEnumerable<string> namespacePrefixes)
+        {
+            using (var context = _contextFactory.CreateContext())
+            {
+                var vendor = await CreateOrGetVendorAsync(userEmail, userName, namespacePrefixes);
+                var user = await context.Users.SingleAsync(u => u.Email.Equals(userEmail));
+
+                user.Vendor = vendor;
+                await context.SaveChangesAsync();
+            }
+        }
+
+        public Vendor CreateOrGetVendor(string userEmail, string userName, IEnumerable<string> namespacePrefixes)
+        {
+            return CreateOrGetVendorAsync(userEmail, userName, namespacePrefixes).GetResultSafely();
```

Comment:
Is this the right time to start using async?  Shouldn't we introduce async to this project in a separate ticket rather than mixing usages?

Reply:
done